app/gpg.go

```diff
@@ -0,0 +1,115 @@
+package app
+
+import (
+	"bytes"
+	"encoding/base64"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"golang.org/x/crypto/openpgp/armor"
+	"golang.org/x/crypto/openpgp/packet"
+)
+
+func FormatPubKey(pubkeyFilePath string) (string, error) {
+	pubKey, err := ioutil.ReadFile(pubkeyFilePath)
+
+	if err != nil {
+		return "Error opening File: ", err
+	}
+
+	if strings.Contains(string(pubKey), "-----BEGIN PGP") {
+		b, err := os.Open(pubkeyFilePath)
+		if err != nil {
+			return "Cannot open file:", err
+		}
+
+		block, err := armor.Decode(b)
+
+		if err != nil {
+			return "Error Decoding: ", err
+		}
+
+		if block.Type == "PGP PUBLIC KEY BLOCK" {
+			reader := packet.NewReader(block.Body)
+			pkt, err := reader.Next()
+			if err != nil {
+				return "Error reading file: ", err
+			}
+
+			_, ok := pkt.(*packet.PublicKey)
+			if !ok {
+				return ("Invalid Public Key"), err
+			}
+		}
+		base64string := base64.StdEncoding.EncodeToString(pubKey)
+		return base64string, nil
+
+	} else {
+		// It's likely a binary file
+		pack, err := packet.Read(bytes.NewReader(pubKey))
+		if err != nil {
+			return "Error reading pub key", err
+		}
+
+		// Was it really a public key file ? If yes, get the PublicKey
+		_, ok := pack.(*packet.PublicKey)
+		if !ok {
+			return "Invalid public key.", err
+		}
+		base64string := base64.StdEncoding.EncodeToString(pubKey)
+		return base64string, err
+	}
+}
+
+func FormatSignature(sigFilePath string) (string, error) {
+	sigKey, err := ioutil.ReadFile(sigFilePath)
+
+	if err != nil {
+		return "Error opening File: ", err
+	}
+
+	if strings.Contains(string(sigKey), "-----BEGIN PGP") {
+		// It's an armored file
+		b, err := os.Open(sigFilePath)
```

Comment:
same as above - you can probably just get away with re-reading `sigKey` instead of re-reading it?